tests/flytekit/unit/annotated/test_type_hints.py

```diff
@@ -477,46 +477,49 @@ def t2(a: str) -> str:
     @workflow
     def my_wf(a: int, b: str) -> (int, str):
         x, y = t1(a=a)
-        d = conditional().if_(x == 4).then(t2(a=b)).elif_(x >= 5).then(t2(a=y)).else_().fail("Unable to choose branch")
-        return x, d
+        d = (
+            conditional("test1")
+            .if_(x == 4)
+            .then(t2(a=b))
+            .elif_(x >= 5)
+            .then(t2(a=y))
+            .else_()
+            .fail("Unable to choose branch")
+        )
+        f = conditional("test2").if_(d == "hello ").then(t2(a="It is hello")).else_().then(t2(a="Not Hello!"))
+        return x, f
 
     x = my_wf(a=5, b="hello ")
-    assert x == (7, "world")
-
-
-def test_wf1_branches_no_else():
-    with pytest.raises(AssertionError):
-
-        def foo():
-            @task
-            def t1(a: int) -> typing.NamedTuple("OutputsBC", t1_int_output=int, c=str):
-                return a + 2, "world"
-
-            @task
-            def t2(a: str) -> str:
-                return a
-
-            @workflow
-            def my_wf(a: int, b: str) -> (int, str):
-                x, y = t1(a=a)
-                d = conditional().if_(x == 4).then(t2(a=b)).elif_(x >= 5).then(t2(a=y))
-                return x, d
-
-            @workflow
-            def my_wf2(a: int, b: str) -> (int, str):
-                x, y = t1(a=a)
-                d = (
-                    conditional()
-                    .if_(x == 4)
-                    .then(t2(a=b))
-                    .elif_(x >= 5)
-                    .then(t2(a=y))
-                    .else_()
-                    .then(t2(a="Ok I give up!"))
-                )
-                return x, d
-
-        foo()
+    assert x == (7, "Not Hello!")
+
+    x = my_wf(a=2, b="hello ")
+    assert x == (4, "It is hello")
+
+
+# def test_wf1_branches_no_else():
```

Comment:
should this be here?

Reply:
i changed it a little bit and it works now